### DISCUSSION TITLE
Documentation for older API versions

### DISCUSSION THREAD

**Discussion by ingmarschuster** [jax-ml/jax#10060] [Q&A]
Hi, 

I've been using `jax.ops.index[x,y,z]` in my code, which broke after updating. However, there seems to be no documentation for old Jax version, which would be very helpful in order to fix my code. Where can I find this?

Yours
Ingmar

**Comment by jakevdp** [comment]
We don't currently maintain versioned documentation, but that particular functionality had a deprecation cycle with a warning for several versions. You can use `jnp.index_exp[x, y, z]` instead – it's an API that's already in numpy itself, so there was no reason to duplicate it elsewhere.

There's a bit more information in the changelog entry discussing this utility's removal: https://github.com/google/jax/blob/main/CHANGELOG.md#jax-032-march-16-2022

**Comment by YouJiacheng** [comment]
FYI, there is the implementation of `jax.ops.index` of [jax==0.2.21](https://files.pythonhosted.org/packages/0f/85/0499931fe8e9dc05f4dd5ef989be2db4653d429adf08b9371fc259402af0/jax-0.2.21.tar.gz)
```python
class _Indexable(object):
  """Helper object for building indexes for indexed update functions.

  This is a singleton object that overrides the :code:`__getitem__` method
  to return the index it is passed.

  >>> jax.ops.index[1:2, 3, None, ..., ::2]
  (slice(1, 2, None), 3, None, Ellipsis, slice(None, None, 2))
  """
  __slots__ = ()

  def __getitem__(self, index):
    return index

#: Index object singleton
index = _Indexable()
```

**Reply by hawkinsp** [reply]
However, as Jake suggests, I would use `jnp.index_exp` or `jnp.s_`.  If I recall correctly `jnp.s_` is exactly equivalent to the old `jax.ops.index`.
